### ISSUE TITLE
Add support for Ruby 3.0

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/manageiq-messaging#70]
I tried adding Ruby 3.0 in #69 , but it fails with 

https://github.com/Fryguy/manageiq-messaging/runs/5856482330?check_suite_focus=true

```
  1) ManageIQ::Messaging::Kafka::Client#initialize converts username/password to sasl parameters
     Failure/Error: @kafka_client = ::Rdkafka::Config.new(rdkafka_connection_opts(options))

       #<Rdkafka::Config (class)> received :new with unexpected arguments
         expected: ({:"bootstrap.servers"=>"localhost:1234", :"client.id"=>"my-ref", :"sasl.password"=>"password", :"sasl.username"=>"user"})
              got: ({:"bootstrap.servers"=>"localhost:1234", :"client.id"=>"my-ref", :"sasl.password"=>"password", :"sasl.username"=>"user"})
     # ./lib/manageiq/messaging/kafka/client.rb:83:in `initialize'
     # ./spec/manageiq/messaging/kafka/client_spec.rb:46:in `new'
     # ./spec/manageiq/messaging/kafka/client_spec.rb:46:in `block (3 levels) in <top (required)>'

  2) ManageIQ::Messaging::Kafka::Client#initialize creates a client connects to a single host
     Failure/Error: @kafka_client = ::Rdkafka::Config.new(rdkafka_connection_opts(options))

       #<Rdkafka::Config (class)> received :new with unexpected arguments
         expected: ({:"bootstrap.servers"=>"localhost:1234", :"client.id"=>"my-ref", :"sasl.protocol"=>"SASL_SSL"})
              got: ({:"bootstrap.servers"=>"localhost:1234", :"client.id"=>"my-ref", :"sasl.protocol"=>"SASL_SSL"})
     # ./lib/manageiq/messaging/kafka/client.rb:83:in `initialize'
     # ./spec/manageiq/messaging/kafka/client_spec.rb:22:in `new'
     # ./spec/manageiq/messaging/kafka/client_spec.rb:22:in `block (3 levels) in <top (required)>'
```

Part of https://github.com/ManageIQ/manageiq/issues/21536

**Comment by kbrock** [comment]
Huh, these errors don't make any sense to me.
Those 2 hashes look the same to me.

Could you share what I missed?

**Comment by chessbyte** [comment]
> Could you share what I missed?

@kbrock https://github.com/Fryguy/manageiq-messaging/runs/5856482330?check_suite_focus=true

**Comment by Fryguy** [comment]
My gut telling me this is a kwargs thing that rspec just doesn't know how to print differently.

**Comment by kbrock** [comment]
TL;DR - a recent change to rspec makes the specs not handle kwargs as gracefully as they did before.
So going forward, I think we'll see a lot more of the now required `{}` around hashes.
